Export `sprintf()` directly

dohse opened the pull request:
Make `require('sprintf')` return a function to enable direct usage like:

```
var sprintf = require('sprintf');
sprintf('Hallo %i. $Welt!', 1);
```

dohse committed:
Export `sprintf()` directly

Make `require('sprintf')` return a function to enable direct usage like:

    var sprintf = require('sprintf');
    sprintf('Hallo %i. $Welt!', 1);

maritz commented:
Sorry for the late reply. This actually breaks the old behaviour and makes vsprintf unavailable.

dohse commented:
`sprintf()` and `vsprintf()` are still be available as `sprintf.sprintf()` and `sprintf.vsprintf()` so no current behavior will be affected. The exported entity will just change from object to function and the two functions will be attached to it.

maritz commented:
Yeah, I tried that with your patch applied, didn't work. I'm not exactly sure on this, but I think you can't use module.exports.xyz and exports.xyz ind the same file.

(There's a slight chance that I did something wrong here, because I'm a bit stupid today. Can you just test it on your end again for all 3 cases?)

dohse committed:
Test exported functions

dohse committed:
Assign to attributes of exported function

dohse commented:
Mea culpa. When exporting a function i need to assign to attributes of the exported function. Should work now - see included test.

maritz commented:
Okay, works. Only problem now is that the test file doesn't actually show anything. Could you either remove it or (ideally) actually write a simple test success/failure measurement and display?

dohse committed:
Convert test to nodeunit

Run it with

    nodeunit test/function-export.js

Install nodeunit with

    npm install -g nodeunit

dohse commented:
I converted the test to nodeunit

dohse commented:
Are there any issues left?

maritz commented:
No, thanks a lot. :)
